### PR TITLE
feat(connector): better names for connector properties

### PR DESCRIPTION
**Pull request by TennyZhuang**
Signed-off-by: TennyZhuang <zty0826@gmail.com>

I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

We've discussed naming style in #4158, but it's tough and needs much effort to decide on every property name. After many trade-offs, I kept every property consistent with FlinkSQL, but renamed "-" to "_" for pg style.

Recently, we posted some tutorials based on the old naming styles. We can update the docs, but someone may play with the latest stable release while others play with the latest main branch, so I decided to keep them compatible for a while. We can remove the deprecated naming later.

The lucky thing is that the PR will not introduce any breaking change, and we can always use a better name without breaking changes. Although the current naming style may not be the best, we can always propose a better name! The attribute macro `#[serde(alias = ?)]` allows multiple aliases for one field so that we can maintain the deprecated names with few efforts.

The PR only changed public APIs, some private APIs such as nexmark, datagen, and all sink connectors are not changed here.

***PLEASE DO NOT LEAVE THIS EMPTY !!!***

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### Types of user-facing changes

* Connector (sources & sinks)

### Release note

* Rename almost all connector properties

We need a detailed doc about the PR, but not here :)

## Refer to a related PR or issue link (optional)

#4158

### CODE REVIEW

**Comment by TennyZhuang** [comment]
To keep the PR clean and can be reviewed by members from other teams, I only update minimal codes. I will update all test codes in later PRs.

**Comment by TennyZhuang** [comment]
All docs, tutorials, and some potential codes used by the cloud team should update the APIs after the next release.

File: src/connector/src/source/kafka/mod.rs [hunk 1]
```diff
@@ -28,19 +28,19 @@ pub const KAFKA_CONNECTOR: &str = "kafka";
 
 #[derive(Clone, Debug, Deserialize)]
 pub struct KafkaProperties {
-    #[serde(rename = "kafka.brokers")]
+    #[serde(rename = "properties.bootstrap.server", alias = "kafka.brokers")]
```
Comment (neverchanje) [inline src/connector/src/source/kafka/mod.rs:31]: I feel 'brokers' is easier to comprehend

Comment (neverchanje) [reply src/connector/src/source/kafka/mod.rs:31]: Also Materialize uses the term 'brokers',

File: src/connector/src/source/kafka/mod.rs [hunk 2]
```diff
@@ -28,19 +28,19 @@ pub const KAFKA_CONNECTOR: &str = "kafka";
...
-    #[serde(rename = "kafka.scan.startup.mode")]
+    #[serde(rename = "scan.startup.mode", alias = "kafka.scan.startup.mode")]
     pub scan_startup_mode: Option<String>,
 
-    #[serde(rename = "kafka.time.offset")]
+    #[serde(rename = "scan.startup.timestamp_millis", alias = "kafka.time.offset")]
     pub time_offset: Option<String>,
 
-    #[serde(rename = "kafka.consumer.group")]
+    #[serde(rename = "properties.group.id", alias = "kafka.consumer.group")]
```
Comment (neverchanje) [inline src/connector/src/source/kafka/mod.rs:43]:
```suggestion
    #[serde(rename = "consumer.group.id", alias = "kafka.consumer.group")]
```

**Comment by neverchanje** [comment]
I dont think we need to completely follow flink's naming convention. It's not our product policy anyway.

File: src/connector/src/source/kafka/mod.rs (hunk 1 above)
Comment (TennyZhuang) [reply src/connector/src/source/kafka/mod.rs:31]: In fact, they are different things, but FlinkSQL only supports `bootstrap.server`.

I'm not sure which one is better, but I don't want to review every name and I decide to follow all FlinkSQL terms :)

File: src/connector/src/source/kafka/mod.rs (hunk 2 above)
Comment (TennyZhuang) [reply src/connector/src/source/kafka/mod.rs:43]: From FlinkSQL too

**Comment by TennyZhuang** [comment]
I agree that we don't need to follow FlinkSQL, but I found that it's even worse when every author has their style. At least, FlinkSQL is a good guide for us currently.

I've found many bad namings. The same term has different names and different styles in different connectors. And designing a new naming style is very boring and tedious. Before we have our own guide, I'd prefer to disallow every author and reviewer using their own style.

**Comment by TennyZhuang** [comment]
Or you can take over to design a complete guide and review every property name. It's very important.

**Comment by TennyZhuang** [comment]
Currently, someone follows kafka-style, someone follows FlinkSQL, and someone follows materialize. It's really hell.

File: src/connector/src/source/kafka/mod.rs (hunk 1 above)
Comment (TennyZhuang) [reply src/connector/src/source/kafka/mod.rs:31]: https://stackoverflow.com/questions/50312579/what-is-difference-between-broker-list-and-bootstrap-servers

**Review by neverchanje — APPROVED** [review]
Alright, I agree that we can first follow Flink's naming style. I don't have an opinion on these proerties right now, because in the future we may prioritize the experience on web UI rather than leading users to the SQL CLI. At last, it won't be in our critical path. So let's keep it simple for now.

**Review by tabVersion — APPROVED** [review]
I support the proposal since we don't have our naming style. At least, we make our interface easier to use.

**Comment by TennyZhuang** [comment]
Merged first, we can discuss about our styles later.

**Pull request merged**